editor/debugger/script_editor_debugger.cpp

```diff
@@ -810,7 +810,9 @@ void ScriptEditorDebugger::_notification(int p_what) {
 
 			const uint64_t until = OS::get_singleton()->get_ticks_msec() + 20;
 
-			while (peer->has_message()) {
+			while (true) {
+				ERR_FAIL_COND_MSG(peer.is_null(), "The debugger peer reference was NULL. (It probably have died).");
```

Comment:
Why not simply:
```diff
diff --git a/editor/debugger/script_editor_debugger.cpp b/editor/debugger/script_editor_debugger.cpp
index 122d38dea3..e2dcd682a9 100644
--- a/editor/debugger/script_editor_debugger.cpp
+++ b/editor/debugger/script_editor_debugger.cpp
@@ -812,7 +812,7 @@ void ScriptEditorDebugger::_notification(int p_what) {
 
                        const uint64_t until = OS::get_singleton()->get_ticks_msec() + 20;
 
-                       while (peer->has_message()) {
+                       while (peer.is_valid() && peer->has_message()) {
 
                                Array arr = peer->get_message();
                                if (arr.size() != 2 || arr[0].get_type() != Variant::STRING || arr[1].get_type() != Variant::ARRAY) {
```

Reply:
to print the error message, does it really need or I just go with that?
@Faless

Reply:
Yeah, I don't think we need the error message there

Reply:
Done. and thanks for the review
@Faless